query/stdlib/testing/end_to_end_test.go

```diff
@@ -222,91 +222,42 @@ func testFlux(t testing.TB, l *launcher.TestLauncher, file *ast.File) {
 		OrganizationID: l.Org.ID,
 		Compiler:       lang.ASTCompiler{AST: bs},
 	}
+
 	if r, err := l.FluxQueryService().Query(ctx, req); err != nil {
 		t.Fatal(err)
 	} else {
+		results := make( map[string]*bytes.Buffer )
+
 		for r.More() {
 			v := r.Next()
-			if err := v.Tables().Do(func(tbl flux.Table) error {
-				return tbl.Do(func(reader flux.ColReader) error {
-					return nil
-				})
-			}); err != nil {
+
+			if _, ok := results[v.Name()]; !ok {
+				results[v.Name()] = &bytes.Buffer{}
+			}
+			err := execute.FormatResult(results[v.Name()], v)
+			if err != nil {
 				t.Error(err)
 			}
 		}
-	}
-
-	// quirk: our execution engine doesn't guarantee the order of execution for disconnected DAGS
-	// so that our function-with-side effects call to `to` may run _after_ the test instead of before.
-	// running twice makes sure that `to` happens at least once before we run the test.
-	// this time we use a call to `run` so that the assertion error is triggered
-	runCalls := stdlib.TestingRunCalls(pkg)
-	pkg.Files[len(pkg.Files)-1] = runCalls
-
-	bs, err = json.Marshal(pkg)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	req = &query.Request{
-		OrganizationID: l.Org.ID,
-		Compiler:       lang.ASTCompiler{AST: bs},
-	}
-	r, err := l.FluxQueryService().Query(ctx, req)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	needInspect := false
-	for r.More() {
-		v := r.Next()
-		if err := v.Tables().Do(func(tbl flux.Table) error {
-			return tbl.Do(func(reader flux.ColReader) error {
-				return nil
-			})
-		}); err != nil {
-			needInspect = true
+		if err := r.Err(); err != nil {
 			t.Error(err)
 		}
-	}
-	if err := r.Err(); err != nil {
-		t.Error(err)
-		needInspect = true
-	}
-	if needInspect {
-		// Replace the testing.run calls with testing.inspect calls.
-		pkg.Files[len(pkg.Files)-1] = inspectCalls
-		bs, err = json.Marshal(pkg)
-		if err != nil {
-			t.Fatal(err)
-		}
-		req = &query.Request{
-			OrganizationID: l.Org.ID,
-			Compiler:       lang.ASTCompiler{AST: bs},
-		}
-		r, err := l.FluxQueryService().Query(ctx, req)
-		if err != nil {
-			t.Fatal(err)
-		}
-		var out bytes.Buffer
-		defer func() {
-			if t.Failed() {
-				scanner := bufio.NewScanner(&out)
+
+		logFormatted := func( name string, results map[string]*bytes.Buffer ) {
+			if _, ok := results[name]; ok {
+				scanner := bufio.NewScanner(results[name])
 				for scanner.Scan() {
-					t.Log(scanner.Text())
+					t.Log( scanner.Text())
 				}
-			}
-		}()
-		for r.More() {
-			v := r.Next()
-			err := execute.FormatResult(&out, v)
-			if err != nil {
-				t.Error(err)
+			} else {
+				t.Log( "table ", name, " not present in results" )
```

Comment:
Hmm....what does it really mean if the `want`, `got` or `diff` is not present in the results? I'm trying to think of the clearest way to express that for troubleshooting purposes.

Reply:
I added this after, when I discovered that "got" could be empty and it would never end up in the formatted results. Then decided to just check them all. If we still have flakiness WRG to the data not getting committed to the DB before the test runs, maybe this happen more often.

Reply:
Yeah, that's a good point. Another way to check for causes of flakiness.